lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -685,12 +685,20 @@ unsafe_to_atom(Part, Line, #elixir_tokenizer{}) when
     is_binary(Part) andalso size(Part) > 255;
     is_list(Part) andalso length(Part) > 255 ->
   {error, {Line, "atom length must be less than system limit: ", [$: | Part]}};
-unsafe_to_atom(Binary, _Line, #elixir_tokenizer{existing_atoms_only=true}) when is_binary(Binary) ->
-  {ok, binary_to_existing_atom(Binary, utf8)};
+unsafe_to_atom(Binary, Line, #elixir_tokenizer{existing_atoms_only=true}) when is_binary(Binary) ->
+  try
+    {ok, binary_to_existing_atom(Binary, utf8)}
+  catch
+    error:badarg -> {error, {Line, "unsafe atom does not exist: ", binary}}
```

Comment:
s/binary/Binary

Reply:
Good catch! Shows the importance of the test. :+1:

Reply:
Thanks for the catch!